### ISSUE TITLE
Conditionally disabled code is not detected if `cfg_attr` predicate is negated

### ISSUE THREAD

**Issue by michcio1234** [intellij-rust/intellij-rust#10501]
## Environment

* **IntelliJ Rust plugin version:** 0.4.194.5382-231
* **Rust toolchain version:** 1.67.1 (d5a82bbd2 2023-02-07) x86_64-unknown-linux-gnu
* **IDE name and version:** PyCharm 2023.1.2 Community Edition (PC-231.9011.38)
* **Operating system:** Linux 5.19.0-41-generic
* **Macro expansion:** enabled

## Problem description
Conditionally disabled code is not detected if `cfg_attr` predicate is negated.

If a predicate is simple, like `feature = "foo"`, then the line with the attribute becomes gray, and a tooltip is shown when hovering the mouse over it:
![image](https://github.com/intellij-rust/intellij-rust/assets/16254622/3d1b4d08-2178-4705-8431-23072cc01a7d)

But if a predicate is negated, like `not(feature = "foo")`, then the line is not gray and the tooltip is not shown:
![image](https://github.com/intellij-rust/intellij-rust/assets/16254622/fb3c0af5-6ebb-4f46-9044-72b3abb80498)

## Steps to reproduce
```rust
// This works:
#[cfg_attr(feature = "foo", ignore)]

// This doesn't work:
#[cfg_attr(not(feature = "foo"), ignore)]
```

**Comment by hydra** [comment]
See also #10518
